### PR TITLE
fix Issue 15660 - break immutable with pure function and mutable refe…

### PR DESCRIPTION
**Pull request by WalterBright**
…rence params

### CODE REVIEW

**Comment by WalterBright** [comment]
Blocked by https://github.com/dlang/phobos/pull/6293

**Review by jacob-carlborg — APPROVED** [review]

File: src/dmd/dcast.d [hunk 1]
```diff
@@ -784,7 +784,7 @@ extern (C++) MATCH implicitConvTo(Expression e, Type t)
 
         override void visit(CallExp e)
         {
-            enum LOG = 0;
+            enum LOG = false;
```
Comment (UplinkCoder) [inline src/dmd/dcast.d]: this is not related to the rest of the pr.

**Comment by WalterBright** [comment]
Blocked by https://github.com/dlang/phobos/pull/6305

**Comment by WalterBright** [comment]
@wilzbach another blocking problem in biguintcore.d

**Comment by wilzbach** [comment]
(merged the Phobos PR and rebased this one to restart the CIs)

**Comment by WalterBright** [comment]
With all the jenkins failures, looks like I'll have to put this change behind -dip1000

**Comment by WalterBright** [comment]
@wilzbach can you please restart the jenkins tests?

**Comment by JinShil** [comment]
I restarted Jenkins

**Comment by wilzbach** [comment]
(These failures aren't spurious, but real breakage.)
